src/dosfsck.c

```diff
@@ -198,6 +198,9 @@ int main(int argc, char **argv)
         qfree(&mem_queue);
     }
 
+    if (!remain_dirty)
+        clean_dirty_flag(&fs);
```

Comment:
This means clearing volume dirty always? if corruptions is found, the fsck with the "-n" terminates without calling this function?

Reply:
fsck with '-n' option, try to clean dirty flag, but failed. (previous version)

All changes including clean dirty_flag are modified on buffer, and fsck call fs_flush(rw) at last, that function determine all buffer to disk or not.
On the contrary, previous version used fs_write_immed() / set_fat_immed() to write to disk immediately even though '-n' option is given. this is fix patch for that

Reply:
I modified back to original but check also rw flag when call clean_dirty_flag()